packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -292,19 +316,28 @@ class BuildIOSArchiveCommand extends _BuildIOSSubCommand {
       projectImageInfoMap: projectInfoMap,
       templateImageDirName: await app.templateAppIconDirNameForImages,
       projectImageDirName: app.projectAppIconDirName);
-    if (usesTemplate) {
-      messageBuffer.writeln('\nWarning: App icon is set to the default placeholder icon. Replace with unique icons.');
-    }
+
+    final ValidationMessage? templateValidationMessage = _createValidationMessage(
+      isValid: !usesTemplate,
+        messageIfValid: 'App icon is not set to the default placeholder image.',
+      messageIfInvalid: 'App icon is set to the default placeholder icon. Replace with unique icons.');
 
     final List<String> filesWithWrongSize = _imageFilesWithWrongSize(
       imageInfoMap: projectInfoMap,
       imageDirName: app.projectAppIconDirName);
-    if (filesWithWrongSize.isNotEmpty) {
-      messageBuffer.writeln('\nWarning: App icon is using the wrong size (e.g. ${filesWithWrongSize.first}).');
-    }
+
+    final ValidationMessage? sizeValidationMessage = _createValidationMessage(
+      isValid: filesWithWrongSize.isEmpty,
+      messageIfValid: 'You do not have incorrectly sized icons.',
```

Comment:
This is too many negatives (it sounds like an error instead of a good thing) and it's not clear what icons it's referencing (the dart side may also have icons.  Can we instead not print the success case?